### PR TITLE
delete commit

### PR DESCRIPTION
**Pull request by ad-47**
Implement delete commit RPC
Fix other function logic

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 1]
```diff
@@ -94,11 +95,11 @@ private boolean commitRepositoryMappingExists(
       Session session, String commitHash, Long repositoryId) {
     String queryString =
         "SELECT count(*) FROM repository_commit rc WHERE rc.commit_hash = :commitHash AND rc.repository_id = :repoId";
-    Query query = session.createQuery(queryString);
+    Query query = session.createSQLQuery(queryString);
     query.setParameter("commitHash", commitHash);
     query.setParameter("repoId", repositoryId);
-    Long count = (Long) query.getSingleResult();
-    return count > 0;
+    BigInteger count = (BigInteger) query.uniqueResult();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: why? given a commit and a repo we should only get single row.

File: backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java [hunk 2]
```diff
@@ -131,24 +133,34 @@ public String setBlobs(Session session, List<BlobExpanded> blobsList, FileHasher
...
+      Status statusMessage;
       switch (blob.getBlob().getContentCase()) {
         case DATASET:
-          processDataset(session, blob, rootTree, fileHasher, getBlobType(blob), componentEntities);
+          processDataset(blob, rootTree, fileHasher, getBlobType(blob), componentEntities);
           break;
         case ENVIRONMENT:
-          throw new NotYetImplementedException(
-              "not supported yet"); // TODO EL/AJ to throw right exceptions
+          statusMessage =
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]: Please do not combine changes which do not make sense together.

Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]: these changes can directly go in a single commit on to the required PR

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 1 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: this is only checking the commit_repository mapping exists or not so just fetch the count.

File: backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java (hunk 2 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]: Should I remove these in other PR or is it ok for now into this?

**Comment by ravishetye** [comment]
Other PR prefered

Ravi Shetye

**Pull request merged**